src/server/Document/Renderers/LiteRenderer.tsx

```
@@ -25,13 +35,17 @@ export default function LitePageRenderer({
         {helmetLinkTags}
         {helmetScriptTags}
         <style dangerouslySetInnerHTML={{ __html: styles }} />
+        <script src="https://polyfill-fastly.io/v3/polyfill.min.js?features=IntersectionObserver" />
```

**Note for reviewers**: We struggled to get `await import('intersection-observer')` working in the view tracking script, likely because of how it is executed (toStringed & then immediately invoked) so we have opted to use a polyfill CDN to provide the intersection observer code for us. (Aside: this polyfill CDN is quite clever as it will not inject polyfill if the feature is already supported on that browser 🤗 )

---

It may worth enquiring with legal as to whether we can use these polyfills in production: https://www.fastly.com/terms 

I don't know if this equates to an acceptable licence?

It also introduces an injection attack vector; admittedly Fasly is probably a fairly trustworthy url but it would be better to manage it ourselves - it would be good to do a Threat Model update around this

---

Do you think we should remove it then?  This was best endeavours to support as many browsers as possible, but we were unable to do it using the await import pattern. 

We chose Fastly since we already use it for Edge servers, but agree - if we do keep it, then we would need to check with Legal & do a Threat Model. In the spirit of trying to get this live, @shayneahchoon shall we remove this, and add a follow up ticket for it / discuss it in our session next week - I can add an item to the Paper doc.

---

(And if we do remove it, do we need to add a `try/catch` around the Intersection Observer, or do we let it silently fail on non-supported browsers?